hazelcast/src/main/java/com/hazelcast/map/impl/DefaultRecordStore.java

```diff
@@ -808,13 +809,17 @@ public boolean merge(Data key, EntryView mergingEntry, MapMergePolicy mergePolic
                 return true;
             }
             // same with the existing entry so no need to map-store etc operations.
-            if (mapServiceContext.compare(name, newValue, oldValue)) {
+            if (mapServiceContext.compare(name, newValue, oldValue)
+                    && mergingEntry.getExpirationTime() == existingEntry.getExpirationTime()) {
```

Comment:
comparing expiration times adding no value here. just updating expiration time of existing record seems sufficient.
WDYT?